### DISCUSSION TITLE
ld: library not found for -lSystem

### DISCUSSION THREAD

**Discussion by danieljprice** [danieljprice/phantom#191] [Show and tell]
Just a short note on how to fix the following problem when compiling with gfortran:
```
ld: library not found for -lSystem
collect2: error: ld returned 1 exit status
```
which occurs on Mac OS after performing a system update. 

The fix is to reinstall the Xcode command line tools for the new OS:
```
xcode-select --install
```
Sometimes you need to wipe the existing command line tools for this to work:
```
rm -rf /Library/Developer/CommandLineTools
Xcode-select --install
```

**Comment by jameswurster** [comment]
If the above method does not work, then add
` FFLAGS= -L/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib`
to the relevant system block in the Makefile.

After implementing the above patch, I could again run & compile Phantom.  However, I could not compile other scripts using gfortran due the same issue.  I added the following alias to my bash_profile to permit me to again compile:
`alias        gfortran='gfortran -L/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib'`

**Comment by giulia-ballabio** [comment]
If the above does not work and you want to avoid adding an additional flag to each Makefile, it is possible that you might still need to upgrade the installed formulae to resolve linking issues. This can be done with Homebrew, so after reinstalling the Xcode Command Line Tools 
`xcode-select --install`
you need to also run
`brew upgrade`
